lib/fizzy/instructions.hpp

```diff
@@ -39,4 +41,12 @@ struct InstructionMetrics
 
 const InstructionMetrics* get_instruction_metrics_table() noexcept;
 
+struct InstructionType
+{
+    std::initializer_list<ValType> inputs;
```

Comment:
I'm still not sure whether it's valid to use `initializer_list` like this.

Declaring `constexpr` `initalizer_list` object works only if it's static local or global, but doesn't compile if it's auto https://godbolt.org/z/iWUg6B

The discussions that I found are not very clear to me, but seem to come to the conclusion that standard doesn't guarantee it to be valid.
https://stackoverflow.com/questions/16063123/is-it-legal-to-declare-a-constexpr-initializer-list-object
https://stackoverflow.com/questions/27496004/why-isnt-stdinitializer-list-defined-as-a-literal-type

Reply:
Just learned that in C++20 `std::vector` can be `constexpr`

Reply:
Found relatively simple solution with a custom struct instead of `initializer_list`, pushing it as a separate commit for now.

Reply:
> Just learned that in C++20 `std::vector` can be `constexpr`, that would help a lot.

I don't think it would help. It is `constexpr` because you can use `new` and `delete` in constexpr evaluation (`consteval`), but for initialization dynamic memory allocation would be needed, so compilation would probably fail. See also `constinit`.